com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -984,39 +989,32 @@ internal void OnClientDisconnectFromServer(ulong clientId)
 
                 // Get the NetworkObjects owned by the disconnected client
                 var clientOwnedObjects = NetworkManager.SpawnManager.GetClientOwnedObjects(clientId);
-                if (clientOwnedObjects == null)
-                {
-                    // This could happen if a client is never assigned a player object and is disconnected
-                    // Only log this in verbose/developer mode
-                    if (NetworkManager.LogLevel == LogLevel.Developer)
-                    {
-                        NetworkLog.LogWarning($"ClientID {clientId} disconnected with (0) zero owned objects!  Was a player prefab not assigned?");
-                    }
-                }
-                else
+
+                // Handle despawn & destroy or change ownership
+                for (int i = clientOwnedObjects.Count - 1; i >= 0; i--)
                 {
-                    // Handle changing ownership and prefab handlers
-                    for (int i = clientOwnedObjects.Count - 1; i >= 0; i--)
+                    var ownedObject = clientOwnedObjects[i];
+                    if (ownedObject != null)
                     {
-                        var ownedObject = clientOwnedObjects[i];
-                        if (ownedObject != null)
+                        if (!ownedObject.DontDestroyWithOwner)
```

Comment:
Just a thought, this is a lot of nested ifs. What are your thoughts on nested ifs vs early exits in this type of Netcode context?

Reply:
Well, I wouldn't exit early because there is other code that needs to run but I did change this to check if it is null and if so then it just continues. 

Nesting doesn't bother me personally and in the end the op codes generated are pretty much identical where it ends up jumping back to the top and grabbing the next element (continue) or returning from the method (exit early).

However, it did remove one layer of a nested if which can be easier to read for users...so that is a good catch.